### PR TITLE
fix: yield reasoning chunks before content chunks in providers

### LINKED ISSUE #1461: [BUG] Assistant message text is truncated when reasoning and content are streamed in the same delta
### **Describe the bug**
When using LLM providers (e.g., LiteLLM, Qwen, NanoGPT, etc.), if the API streams both `reasoning_content` (or equivalent thinking tags) and `content` within the exact same streaming chunk (Delta), the assistant's message text gets truncated in the UI. This happens because the system processes and commits the text content first, causing the internal task logic to get tangled when reasoning arrives immediately after in the same chunk.

### **To Reproduce**
1. Connect to an API provider/proxy that batches streaming chunks (e.g., LiteLLM) or a model that sends reasoning and content simultaneously.
2. Trigger a response that contains both a thinking process (`reasoning_content`) and a final answer (`content`).
3. If the network delivers both fields in a single `delta` object, observe the assistant's response in the chat UI.
4. See error: The text content gets unexpectedly cut off (truncated) or fails to render properly.

### **Expected behavior**
The system should strictly yield and process the `reasoning` chunk *before* the `text` chunk whenever both are present in the same delta. This ensures the reasoning process is fully rendered and completed before the actual text content begins, preventing any UI truncation.

### **What version of zoo are you running**
Latest version (main branch)

### **Additional context**
- The issue originates from the `for await (const chunk of stream)` logic in OpenAI-compatible providers (`src/api/providers/*.ts`).
- By default, most providers were yielding `delta.content` before `extractReasoningFromDelta(delta)`.
- Reordering the yield sequence to process `reasoning` first fixes the truncation bug completely. Test cases verifying this exact simultaneous delta scenario have also been written and validated.

### PR DESCRIPTION
**Pull request by daewoongoh**
### Related GitHub Issue

Closes: #1461 

### Description

This PR fixes a critical bug where the assistant's message text gets truncated in the UI when an LLM provider streams both `reasoning_content` (or thinking tags) and final `content` simultaneously in the exact same delta chunk. 

**Key implementation details:**
- Modified the `for await (const chunk of stream)` logic across all OpenAI-compatible providers (`lite-llm`, `qwen-code`, `mimo`, `nanogpt`, `requesty`, `kenari`, `unbound`, `lm-studio`, `opencode-go`, `openai`, `deepseek`, and the `base-openai-compatible-provider`).
- Reordered the yield sequence so that `extractReasoningFromDelta(delta)` is always yielded and processed *before* `delta.content`.
- This ensures the UI accurately renders the complete reasoning process before starting the text rendering, which resolves the internal task truncation issue.

### Test Procedure

- **Unit Tests added:** Added explicit edge-case tests in `lite-llm.spec.ts`, `openai.spec.ts`, and `base-openai-compatible-provider.spec.ts`.
  - These tests mock an `asyncStreamFrom` where `reasoning_content` and `content` are tightly packed in the exact same `delta` object.
  - Asserted strictly that `expect(contentChunks)` yields the `reasoning` object before the `text` object.
- **Run tests:** `pnpm test` (Verified that all 7,935 tests pass with 100% success rate across the repository).

### Pre-Submission Checklist

- [x] **Issue Linked**: This PR is linked to an approved GitHub Issue (see "Related GitHub Issue" above).
- [x] **Scope**: My changes are focused on the linked issue (one major feature/fix per PR).
- [x] **Self-Review**: I have performed a thorough self-review of my code.
- [x] **Testing**: New and/or updated tests have been added to cover my changes (if applicable).
- [x] **Visual Snapshot** (UI changes only): If a user would notice this change at a glance (layout, theme tokens, brand elements, empty/error states), I've added or updated a `*.visual.tsx` snapshot in `webview-ui/`. See `webview-ui/AGENTS.md` → "When a UI change needs a snapshot".
- [x] **Documentation Impact**: I have considered if my changes require documentation updates (see "Documentation Updates" section below).
- [x] **Contribution Guidelines**: I have read and agree to the [Contributor Guidelines](/CONTRIBUTING.md).

### Documentation Updates

- [x] No documentation updates are required.
- [ ] Yes, documentation updates are required. (Please describe what needs to be updated or link to a PR in the docs repository).

### Get in Touch

hehegwk_23849

### CODE REVIEW

**Comment by daewoongoh** [comment]
@coderabbitai review

File: src/api/providers/lm-studio.ts [hunk 1]
```diff
@@ -139,6 +132,13 @@ export class LmStudioHandler extends BaseProvider implements SingleCompletionHan
 					yield { type: "reasoning", text: reasoningText }
```
Comment (edelauna) [inline src/api/providers/lm-studio.ts]: Does `lmstudio.spec.ts` have a test where both `reasoning_content` and `content` arrive in the same delta and the ordering is asserted? Without one, a revert here would pass all existing tests.

**Review by edelauna — APPROVED** [review]
